src/Model/DocumentModel.cpp

```diff
@@ -0,0 +1,147 @@
+#include "DocumentModel.hpp"
+
+#include "Utils/Utils.hpp"
+
+constexpr auto const_class_name = "class";
+
+DocumentModel::DocumentModel(const nlohmann::json& schema, const nlohmann::json& document)
+{
+  m_validator.setRootSchema(schema);
+  if (m_validator.validate(document))
+  {
+    from_json(document, m_doc);
+  }
+  else
+  {
+    throw std::runtime_error("Invalid document");
```

Comment:
It is not a good practice to throw exceptions in constructor.

Reply:
Fixed, remove validating, move up validator setup.